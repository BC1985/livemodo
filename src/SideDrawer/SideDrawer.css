.side-drawer {
  height: 100%;
  background-color: rgb(27, 27, 31);
  box-shadow: 20px 0px 20px 2px rgba(0, 0, 0, 0.5);
  position: fixed;
  top: 0;
  left: 0;
  width: 50%;
  max-width: 400px;
  z-index: 200;
}
.side-drawer.open {
  transform: translateX(0);
  transition: 1s;
}

.side-drawer ul {
  list-style: none;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.side-drawer li {
  margin-bottom: 20px;
}

.side-drawer a {
  color: rgb(184, 179, 179);
  text-decoration: none;
  font-size: 1.2rem;
}

.side-drawer a:hover,
.side-drawer a:active {
  cursor: pointer;
  color: orangered;
}

/* ----------- iPad 3, 4 and Pro 9.7" ----------- */

/* Portrait */
@media only screen and (min-device-width: 768px) and (max-device-width: 1024px) and (orientation: portrait) and (-webkit-min-device-pixel-ratio: 2) {
  .side-drawer.open {
    display: none;
  }
}

/* Landscape */
@media only screen and (min-device-width: 768px) and (max-device-width: 1024px) and (orientation: landscape) and (-webkit-min-device-pixel-ratio: 2) {
  .side-drawer.open {
    display: none;
  }
}
/* ----------- iPad Pro 10.5" ----------- */

/* Portrait and Landscape */
@media only screen and (min-device-width: 834px) and (max-device-width: 1112px) and (-webkit-min-device-pixel-ratio: 2) {
  .side-drawer.open {
    display: none;
  }
}

/* ----------- iPad Pro 12.9" ----------- */

/* Portrait and Landscape */
@media only screen and (min-device-width: 1024px) and (max-device-width: 1366px) and (-webkit-min-device-pixel-ratio: 2) {
  .side-drawer.open {
    display: none;
  }
}

/* Laptops */
/* ----------- Non-Retina Screens ----------- */
@media screen and (min-device-width: 1200px) and (max-device-width: 1600px) and (-webkit-min-device-pixel-ratio: 1) {
  .side-drawer.open {
    display: none;
  }
}

/* ----------- Retina Screens ----------- */
@media screen and (min-device-width: 1200px) and (max-device-width: 1600px) and (-webkit-min-device-pixel-ratio: 2) and (min-resolution: 192dpi) {
  .side-drawer.open {
    display: none;
  }
}
@media screen and (min-device-width: 1600px) {
  .side-drawer.open {
    display: none;
  }
}
/* ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ */
