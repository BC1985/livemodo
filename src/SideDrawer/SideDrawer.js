import React from "react";
import "./SideDrawer.css";
import { NavLink } from "react-router-dom";

const sideDrawer = props => {
  let drawerClasses = "side-drawer";
  if (props.show) {
    drawerClasses = "side-drawer open";
  }
  return (
    <nav className={drawerClasses}>
      <ul>
        {props.routes.map(route => (
          <li key={route.name}>
            <NavLink to={route.to} onClick={route.onClick}>
              {route.name}
            </NavLink>
          </li>
        ))}
      </ul>
    </nav>
  );
};

export default sideDrawer;
