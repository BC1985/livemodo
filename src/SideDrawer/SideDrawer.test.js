import React from "react";
import ReactDOM from "react-dom";
import SideDrawer from "../SideDrawer/SideDrawer";
import { MemoryRouter, NavLink } from "react-router-dom";
const routes = [
  {
    to: "/",
    name: "Home"
  },
  {
    to: "/browse",
    name: "Browse Reviews"
  },
  {
    to: "/login",
    name: "Log in"
  }
];
it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <MemoryRouter>
      <SideDrawer routes={routes} />
    </MemoryRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});
