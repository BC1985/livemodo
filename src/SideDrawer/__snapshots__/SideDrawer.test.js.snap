// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<nav
  className="side-drawer"
>
  <ul>
    <li>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        Home
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/browse"
        onClick={[Function]}
      >
        Browse Reviews
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/login"
        onClick={[Function]}
      >
        Log in
      </a>
    </li>
  </ul>
</nav>
`;
